olsr-routing-protocol.cc

```diff
@@ -1820,11 +1833,34 @@ RoutingProtocol::SendTc ()
   tc.ansn = m_ansn;
   
   //TODO: MODIFICATION 3: Randomly remove half the elements from MprSelectorSet
-  for (MprSelectorSet::const_iterator mprsel_tuple = m_state.GetMprSelectors ().begin ();
-       mprsel_tuple != m_state.GetMprSelectors ().end (); mprsel_tuple++)
-    {
-      tc.neighborAddresses.push_back (mprsel_tuple->mainAddr);
+  
+   if(m_mode == 3) {
```

Comment:
This would be much easier to follow if m_mode were an enum

Reply:
I hope I don't have to to write much network interference code again